driver.go

```diff
@@ -325,23 +386,35 @@ func (d *Driver) RecoverTask(handle *drivers.TaskHandle) error {
 	}
 	d.logger.Debug("Checking for recoverable task", "task", handle.Config.Name, "taskid", handle.Config.ID, "container", taskState.ContainerID)
 
-	inspectData, err := d.podman.ContainerInspect(d.ctx, taskState.ContainerID)
-	if errors.Is(err, api.ContainerNotFound) {
-		d.logger.Debug("Recovery lookup found no container", "task", handle.Config.ID, "container", taskState.ContainerID, "error", err)
-		return nil
-	} else if err != nil {
-		d.logger.Warn("Recovery lookup failed", "task", handle.Config.ID, "container", taskState.ContainerID, "error", err)
-		return nil
+	// TODO: It can be created by any podman client, so we need to iterate over all podman clients and find that container back...
+	var podmanClientName string
+	var taskPodmanClient *api.API
+	var inspectData api.InspectContainerData
+	var err error
+	for name, podmanClient := range d.podmanClients {
```

Comment:
When we recover a task, don't we have the task config handy and so we should be able to use that to figure out which client to use, rather than iterating?

Reply:
Yup, I figured it out. In `StartTask()` I found that you can decode the TaskHandle with `DecodeDriverConfig()` to get a handle on a more specific struct that contains all the task fields (incl. the socket name). I did rename this struct to `podmanTaskConfig` in StartTask (and RecoverTask) because it's more clear than `driverConfig`.